### ISSUE TITLE
Provisioned Solr instances are not available

### ISSUE THREAD

**Issue by mogul** [GSA/data.gov#3071]
Newly provisioned Solr instances get stuck and never fully provision.

## How to reproduce

1. Description of steps to reproduce the issue.

- `cf create-service solr-cloud base solr-test`
- `watch cf service solr-test`

## Expected behavior

- The information says `create in progress` for a while, but then
- The information says `create succeeded`

## Actual behavior

- The information says `create in progress` forever

## Sketch
[Notes or a checklist reflecting our understanding of the selected approach]

The Solr (and Zookeeper) instances that are provisioned are stuck in a `Pending` state in AWS Fargate because PersistentVolumeClaims (PVCs) can't be satisfied:

```
  - lastProbeTime: null
    lastTransitionTime: "2021-04-01T23:00:55Z"
    message: 'Pod not supported on Fargate: volumes not supported: data not supported
      because: PVC data-solr-595266624de0c9cc-solrcloud-zookeeper-0 not bound'
    reason: POD_UNSUPPORTED_ON_FARGATE
    status: "False"
    type: PodScheduled
```

Potential alternatives:
* We can [configure EFS storage for our EKS clusters](https://aws.amazon.com/blogs/containers/running-stateful-workloads-with-amazon-eks-on-aws-fargate-using-amazon-efs/) so that PVCs can be satisfied, not just for SolrCloud instances but for workloads generally. 
* We can [investigate the Ephemeral option](https://github.com/apache/solr-operator/blob/main/docs/solr-cloud/solr-cloud-crd.md#data-storage)
  * For now we can use [the Kubernetes `emptyDir` volume type](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir), which is explicitly supported by the operator.
  * We can  also investigate the [ "generic ephemeral volumes" and "storage capacity tracking" features](https://kubernetes.io/blog/2021/04/14/local-storage-features-go-beta/) that hit beta in Kubenetes 1.21 in April 2021.

**Comment by mogul** [comment]
I've got [a PR in flight](https://github.com/GSA/datagov-brokerpak/pull/17) that I think should address this.

More on the operator-provided options here:
https://github.com/apache/solr-operator/blob/main/docs/solr-cloud/solr-cloud-crd.md#data-storage

I did a bunch of research on SolrCloud architecture, and for our purposes, I believe the "ephemeral" `emptyDir` volume type should be fine:
* Up to 20GB of storage is available to each Fargate task.
* Since it's local node storage, performance is about as good as we can ask for from Fargate.
* `emptyDir` storage persists over crashes/restarts of the pod it is associated with. 
* SolrCloud stores multiple replicas of each shard, spread across the SolrCloud nodes.
* The solr-operator cycles SolrCloud nodes carefully so that there's always at least one replica of every shard.
* If a pod gets moved to another host, it immediately gets filled up with replicas from the other SolrCloud nodes holding under-replicated shards.

Making this change to our provisioned SolrCloud instances saves us the effort of implementing Persistent Volumes on our k8s clusters until we have a workload that really needs them. (Note that implementing use of `SolrBackup` can happen without that effort as well.)
